fix: clear timeout in collection operations to prevent memory leak

techcodie opened the pull request:
- Add clearTimeout() calls in success and error paths
- Prevents setTimeout references from accumulating
- Fixes memory leak in high-throughput applications
- Add test to verify timeout cleanup behavior

Fixes memory leak where buffered collection operations would create setTimeout timers that were never cleared when operations completed successfully, causing continuous memory growth.

<!-- Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. The two fields below are mandatory.

If you're making a change to documentation, do **not** modify a `.html` file directly. Instead, find the corresponding `.pug` file or test case in the `test/docs` directory. -->

**Summary**

<!-- Explain the **motivation** for making this change. What problem does the pull request solve? -->

**Examples**

<!-- If this code fixes a bug or adds a new feature, provide an example demonstrating the change, unless you added a test. -->

vkarpov15 reviewed (changes requested):
clearTimeout looks good but there's some unrelated changes that are incorrect

vkarpov15 commented on lib/drivers/node-mongodb-native/connection.js:
```diff
@@ -160,19 +159,21 @@ NativeConnection.prototype.aggregate = function aggregate(pipeline, options) {
 NativeConnection.prototype.removeDb = function removeDb(name) {
-  const dbs = this.otherDbs.filter(db => db.name === name);
-  if (!dbs.length) {
+  const db = this.relatedDbs[name];
```
Undo this change

techcodie commented:
Hi @vkarpov15! Thank you for the detailed feedback. I've addressed all your comments:

✅ Reverted all unrelated otherDbs/relatedDbs changes in connection.js back to original implementation
✅ Removed separate test/collection.timeout.test.js and moved tests to collection.test.js
✅ PR now contains only the setTimeout memory leak fix with proper clearTimeout() calls

The PR is now focused and clean. I really appreciate your guidance and would love to get this memory leak fix MERGED!

techcodie commented:
@vkarpov15  I have resolved all the issues and rechecked them please let me know if something is wrong or missing I will update it, and if it's not can you merge it please.
